### ISSUE TITLE
proposal: noop_primitive_operations allows empty string interpolations

### ISSUE THREAD

**Issue by matanlurey** [dart-lang/linter#4700]
Would it be reasonable to allow the following pattern:

```dart
const message = '' // LINT
  'Warning: This is a test message.\n'
  '\n'
  'Next time, it won't be a test!';
```

Basically, using a leading empty string literal for dart format compliance :)

**Comment by bwilkerson** [comment]
@munificent I don't know whether we've considered updating the formatter to not require including an empty string as part of the planned rework, but it would be nice if this pattern wasn't necessary in order to get the strings to line up.

**Comment by munificent** [comment]
> I don't know whether we've considered updating the formatter to not require including an empty string as part of the planned rework

It's definitely not part of the plan. But I'm open to looking into it. Does anyone who cares about this enough want to [file an issue](https://github.com/dart-lang/dart_style/issues) to request it? There's no way I'll get to it until after the rewrite is done, but afterwards I wouldn't mind looking into it if I can find the time and if other people want it.

**Comment by matanlurey** [comment]
I am going to take a shot at modifying the current lint.
